File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        for num in range(n, n + 11):
            product = 1
            temp = num
            
            # For 0, the product of digits is 0
            if temp == 0:
                product = 0
                
            while temp > 0:
                product *= temp % 10
                temp //= 10
                
            if product % t == 0:
                return num
                
        return -1

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.c
#include <stdio.h>

int smallestNumber(int n, int t) {
    for (int num = n; num <= n + 10; num++) {
        int product = 1;
        int temp = num;
        
        if (temp == 0) {
            product = 0;
        }
        
        while (temp > 0) {
            product *= temp % 10;
            temp /= 10;
        }
        
        if (product % t == 0) {
            return num;
        }
    }
    return -1; 
}
